src/panels/lovelace/cards/hui-conditional-card.ts

```diff
@@ -0,0 +1,75 @@
+import { html, LitElement } from "@polymer/lit-element";
+
+import computeCardSize from "../common/compute-card-size.js";
+import createCardElement from "../common/create-card-element.js";
+import { HomeAssistant } from "../../../types.js";
+import { LovelaceCard, LovelaceConfig } from "../types.js";
+
+interface Condition {
+  entity: string;
+  state?: string;
+  state_not?: string;
+}
+
+interface Config extends LovelaceConfig {
+  card: LovelaceConfig;
+  conditions: Condition[];
+}
+
+class HuiConditionalCard extends LitElement implements LovelaceCard {
+  protected hass?: HomeAssistant;
+  protected config: Config;
+  protected card: LovelaceCard;
+
+  static get properties() {
+    return {
+      hidden: {},
+    };
+  }
+
+  protected render() {
+    return html`
+      <style>
+        .hidden {
+        display: none;
+        }
+      </style>
+      <div id="card"
+      class=${this.hidden ? "hidden" : ""}>
+        ${this.card}
+      </div>
+    `;
+  }
+
+  public setConfig(config) {
+    this.config = config;
+    this.card = createCardElement(config.card);
+    this.card.hass = this.hass;
+  }
+
+  set hass(hass) {
+    this.card.hass = hass;
+
+    const visible = this.config.conditions.every((c) => {
+      if (!(c.entity in hass.states)) return false;
+      if (c.state) return hass.states[c.entity].state === c.state;
+      return hass.states[c.entity].state !== c.state_not;
+    });
+
+    this.hidden = !visible;
```

Comment:
I think that we should change this logic. We can do this line and the next one way more easier by using the CSS `:host` selector. This selector allows styling the host element.

So in our CSS, instead of `.hidden`, we do `:host(.hidden)`. By setting the display of the host to `none`, we also don't need to kill the margin, as it's hidden with it.

After that we just have to set the class correctly

```js
this.classList.toggle('hidden', !visible);
```

Note that you could also prefer an attribute over a class, I don't think it really matters.

btw, it would probably make sense if we detach the card from the DOM and don't update it's `hass` attribute either. That way we're not doing computations for nothing.

And that makes me wonder if we even need LitElement or that this can directly extend HTMLElement and just do `appendChild` and `removeChild` calls combined with a `this.style.display = visible ? null : 'none'

Reply:
Normally `.column >*` from `hui-view` takes precedence over `:host(.hidden)` in this case, but just typing that sentence out made me realize I could add `!important` and fix that issue.

Reply:
And about HTMLElement... well...pfft... sure... I **guess**... if you want to do things the **easy** way... \*grumble

Reply:
But why remove and reappend it?

Reply:
remove when we're hiding it, append when we're showing it

Reply:
but maybe I'm just overthinking this and we can just keep it Lit.

Reply:
Oh you're right, if we're hiding it, we're good too.